### ISSUE TITLE
Group router always drops first message 

### ISSUE THREAD

**Issue by AlexanderNurmsalu** [akka/akka#31297]
Hello,

I have a cluster setup of 2 nodes. One node has service registered with services key over receptionist. Another node is using a group router to contact service on the first node.

[Group router documentation](https://doc.akka.io/docs/akka/current/typed/routers.html#group-router) has an example with the statement:
```
// the group router will stash messages until it sees the first listing of registered services from the 
// receptionist, so it is safe to send messages right away
for (int i = 0; i < 10; i++) {
    router.tell(new Worker.DoLog("msg " + i));
}
```
It states that it is safe to send messages immediately as they are stashed. But it behaves differently.
Whenever a new group router is created, a new/first message gets dropped by the group router that is sent to the service right after the group router creation. 

I also noted that subscriptions to the receptionist return 0 service instances on the first listing provided, while the node with the service is actually up and running. Only the second listing notification provides correct information. Seems like there is something to do with the receptionist and probably listing notifications.

Scala binary version 2.13
Akka version 2.6.19

Br,
Alexander

**Comment by johanandren** [comment]
Yes, there will be a race condition with the local receptionist having seen the routees on the other node when the router subscribes or not, if not the listing will be empty. 

I think we could maybe improve the group router to stay in the initial stashing mode until it sees a listing with at least one routee. It may need a timeout to not keep a message around forever in a scenario where there is no routees for a long while though, a routee arriving an hour after the router was started getting that stashed message would be quite surprising.

**Comment by AlexanderNurmsalu** [comment]
But why do we get a dummy notification with 0 service instances from the receptionist? At the time of subscribing for the listing, the service is up and running. I believe the expectation would be to get the correct list of service instances at the current state of the cluster.
